### ISSUE TITLE
REEExtractor capacity limiter bounds only the newly extracted REE, so organic loading can exceed the extractant inventory

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#208]
## Summary

The stage capacity limiter in `REEExtractor.__call__` applies its smooth saturation to the *increment* extracted at the stage, then adds the REE already on the entering solvent back on top, uncapped. The total organic loading is therefore not bounded by the extractant inventory: `theta_total` can exceed 1, approaching `theta_solvent + 1`.

Pre-existing on `main` (introduced in `ef99177`, refined by #191/#193); not from #207. Found while reviewing #207, which is where it starts to matter — see "Why it surfaces now".

## The code

`src/difflow_ree/units/extraction.py:796-813`:

```python
total_newly_extracted = sum(
    extract_flows[elem] - jnp.asarray(solvent_flows.get(elem, 0.0))
    for elem in p.elements
)
total_newly_extracted = jnp.maximum(total_newly_extracted, 0.0)
max_capacity = F_extractant / self._isotherm.m
scale = _soft_saturation(total_newly_extracted, max_capacity, k)
for elem in p.elements:
    F_solvent_elem = jnp.asarray(solvent_flows.get(elem, 0.0))
    newly_extracted = extract_flows[elem] - F_solvent_elem
    extract_flows[elem] = F_solvent_elem + newly_extracted * scale
```

`_soft_saturation` does guarantee `total_newly_extracted * scale < max_capacity` strictly, so the *increment* is bounded. But the organic outlet carries `F_solvent_elem + newly_extracted * scale`, so

```
theta_total = theta_solvent + m * (newly * scale) / F_extractant  <  theta_solvent + 1
```

and it exceeds 1 for any `theta_solvent > 0` once the stage is anywhere near capacity. The comment at `extraction.py:793-795` states the opposite guarantee:

> It also guarantees `total * scale < capacity` strictly, so the extractant balance never binds more extractant than was fed.

That conclusion holds for the increment, not for the total, which is what the extractant balance actually sees.

## Numbers

With the default `capacity_sharpness = 8` and a solvent entering 40% loaded (`theta_solvent = 0.4`):

| uncapped / capacity | `capacity_scale` | increment contribution to theta | `theta_total` |
|---|---|---|---|
| 0.89 | 0.960 | 0.851 | **1.25** |
| 1.50 | 0.664 | 0.995 | **1.40** |
| 3.00 | 0.333 | 1.000 | **1.40** |

The first row is the uncomfortable one: `capacity_scale = 0.96` reads as "barely limiting, nowhere near the wall", and the organic still leaves holding 25% more REE than the extractant can bind.

`theta_total` *is* reported in `capacity_info`, so the violation is observable after the fact — nothing enforces it, and no test asserts `theta_total <= 1`.

## Why it surfaces now

`free_fraction_in` already reduces `E` as the entering solvent loads up, and deliberately never reaches zero (#193, to keep the gradient alive on the recycled-solvent path). That is the right call, but it means the free-fraction term cannot bound the total by itself — bounding it is exactly the post-loop limiter's job, and that is the part that misses the incoming loading.

With an open circuit the solvent always entered REE-free (`theta_solvent = 0`), where the current form is exactly correct, which is why this has not bitten before. #207 closes the organic loop (#202), so loaded solvent is recycled into the extraction section on every pass — precisely the case the limiter gets wrong, and the case where an over-loaded organic then propagates around the loop.

## Suggested fix

Saturate the total organic loading rather than the increment:

```python
F_ree_solvent_total = sum(
    jnp.asarray(solvent_flows.get(e, 0.0)) for e in p.elements
)
total_organic = total_newly_extracted + F_ree_solvent_total
scale = _soft_saturation(total_organic, max_capacity, k)
for elem in p.elements:
    F_solvent_elem = jnp.asarray(solvent_flows.get(elem, 0.0))
    extract_flows[elem] = (
        F_solvent_elem + (extract_flows[elem] - F_solvent_elem)
    ) * scale
    raffinate_flows[elem] = (
        jnp.asarray(feed_flows.get(elem, 0.0))
        + F_solvent_elem
        - extract_flows[elem]
    )
```

Scaling the total means a solvent entering above capacity gives REE back to the aqueous phase, which is the physically right answer (an over-loaded organic back-extracts) and stays mass-consistent: the existing `raffinate = feed + solvent - extract` line absorbs the difference with no extra bookkeeping. It also keeps every property #193 wanted — smooth, strictly positive, no dead lever.

`theta_solvent = 0` reproduces the current behaviour exactly, so open-loop results and their tests should be unchanged; only loaded-solvent cases move.

Worth adding alongside:

- a test asserting `info["theta_total"] <= 1` (within the smooth limiter's tolerance) for a solvent entering at `theta_solvent` of 0, 0.4 and 1.2;
- correcting the comment at `extraction.py:793-795` to say what is bounded.

## Relationship to #196

The comment at `extraction.py:710-712` already notes that converging a single mass-action extractant balance subsumes all of this, and #196 (merged in #207) adds that path — `MassActionSection` gets loading right by construction, since the extractant is a conserved component. This issue is about the correlation-based `REEExtractor` path, which remains the default and is what the module library and most existing flowsheets use. If the plan is for mass-action to become the default, this is still worth fixing in the meantime rather than leaving a silently violated capacity constraint on the path people are on today.
